base_serializer.go

```diff
@@ -384,3 +384,33 @@ func typeQualifier(ionValue hashValue) byte {
 	typeCode := byte(ionValue.ionType())
 	return typeCode << 4
 }
+
+func compareBytes(bytes1, bytes2 []byte) int {
```

Comment:
Can we add a comment explaining what the meaning of less than between 2 byte arrays means here. Especially for the case when the 2 input arrays have different length. Also which condition takes precedence (value of byte vs size of array)